src/test/java/org/elasticsearch/test/store/MockDirectoryHelper.java

```diff
@@ -122,11 +127,26 @@ public FsDirectoryService randomDirectorService(IndexStore indexStore) {
         private final boolean crash;
         private volatile RuntimeException closeException;
         private final Object lock = new Object();
+        private final Set<String> superUnSyncedFiles;
+        private final Random superRandomState;
 
         public ElasticsearchMockDirectoryWrapper(Random random, Directory delegate, ESLogger logger, boolean crash) {
             super(random, delegate);
             this.crash = crash;
             this.logger = logger;
+
+            // TODO: remove this silliness on Lucene upgrade:
+            try {
+                Field field = MockDirectoryWrapper.class.getDeclaredField("unSyncedFiles");
+                field.setAccessible(true);
+                superUnSyncedFiles = (Set<String>) field.get(this);
+
+                field = MockDirectoryWrapper.class.getDeclaredField("randomState");
+                field.setAccessible(true);
+                superRandomState = (Random) field.get(this);
```

Comment:
Just so I understand, why is it necessary to use the `MockDirectoryWrapper`'s `Random` instead of the `Random` passed in to the `ElasticsearchMockDirectoryWrapper`? Does this mean that they are two different instances and might not follow the same seed?